nvm.sh

```diff
@@ -1983,7 +1983,16 @@ nvm_install_binary_extract() {
     else
       tar='tar'
     fi
-    command "${tar}" -x${tar_compression_flag}f "${TARBALL}" -C "${TMPDIR}" --strip-components 1 || return 1
+    # Special workaround for OpenBSD tar which does not support xz compression  or --strip-components
+    if [ "${NVM_OS}" = 'openbsd' ]; then
+      if [ "${tar_compression_flag}" = 'J' ]; then
+        command xzcat "${TARBALL}" | "${tar}" -xf - -C "${TMPDIR}" -s '/[^\/]*\///' || return 1
```

Comment:
will xzcat _always_ exist on openbsd? like, it's installed by default, on every version?

Reply:
This is a great point that I completely overlooked. The other commenter mentioned it and I have it, so I assumed it was in base, but it's actually part of the `xz` package. That package happens to get pulled in as a dependency by the `python` package on OpenBSD, so I don't think that I've ever worked on a system that didn't already have it, but it's not _strictly_ guaranteed to be on an OpenBSD system. That should probably be incorporated into the function that determines if we download a `.tar.xz` or a `.tar.gz` tarball.

Reply:
In that case, yes, let's detect it, and if it's not present, maybe we need to avoid downloading the xz version in the first place?

Reply:
Ah, double-checking your existing code shows you already do a check for if we have the `xz` command on our PATH. That's provided by the same package in OpenBSD, so you've already handled that.

Reply:
ok great, so if `xz` exists on openbsd, then `xzcat` must as well?

Reply:
Barring something insane, like having installed the package and manually deleted that binary, yes.